qemu/tests/tpm_with_tss.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from virttest import error_context
+from virttest import utils_package
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the TPM device info inside guest and host.
+    Steps:
+        1. Launch a guest with vTPM device
+        2. Clone the tpm2-tss repository and compile it
+        3. Execute the tpm2-tss unit test and analyze the test result
+
+    :param test: QEMU test object.
+    :type  test: avocado.core.test.Test
+    :param params: Dictionary with the test parameters.
+    :type  params: virttest.utils_params.Params
+    :param env: Dictionary with test environment.
+    :type  env: virttest.utils_env.Env
+    """
+
+    configure_cmd = params["configure_cmd"]
+    check_log_cmd = params["check_log_cmd"]
+    make_check_cmd = params["make_check_cmd"]
+    required_packages = params.objects("required_pkgs")
+    tpm_device = params["tpm_device"]
+    tpm2_tss_path = params["tpm2_tss_path"]
+    tpm2_tss_repo = params["tpm2_tss_repo"]
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    error_context.context("Check if TPM2 device exists", test.log.info)
+    if session.cmd_status("test -c %s" % tpm_device) != 0:
```

Comment:
Why use test cmd instead of ls cmd check device here?

Reply:
`ls` can only output the device if it exists, but `test` can check its device type.